exporters/otlp/otlp_test.go

```diff
@@ -104,17 +118,23 @@ func newExporterEndToEndTest(t *testing.T, additionalOpts []otlp.ExporterOption)
 	spans := mc.getSpans()
 
 	// Now verify that we received all spans.
-	if got, want := len(spans), m; got != want {
+	if got, want := len(spans), m*2; got != want {
 		t.Fatalf("span counts: got %d, want %d", got, want)
 	}
 	for i := 0; i < 4; i++ {
 		if gotName, want := spans[i].Name, "AlwaysSample"; gotName != want {
 			t.Fatalf("span name: got %s, want %s", gotName, want)
 		}
-		if got, want := spans[i].Attributes[0].IntValue, int64(i); got != want {
-			t.Fatalf("span attribute value: got %d, want %d", got, want)
+		if got, want := spans[i].Attributes[0].IntValue, m; got > int64(want) {
```

Comment:
I'm not entirely sure why we switched to checking maximal value instead of equality.
Might be worth adding a comment here to explain why.

Reply:
mc.getSpans() contains m*2 spans and the attributes value ranges from 0 to m-1.  The want should have been 'm-1'.
Anyway, I have changed the test to verify that all spans with different attribute values are received for each resourceSpan.
PTAL.

Reply:
I still need to add the verification.